src/chat.rs

```diff
@@ -980,8 +982,17 @@ impl Chat {
             && !self.is_device_talk()
             && !self.is_mailing_list()
             && !self.is_contact_request()
-            && (self.typ == Chattype::Single
-                || is_contact_in_chat(context, self.id, DC_CONTACT_ID_SELF).await?))
+            && self.can_edit(context).await?)
+    }
+
+    /// Checks if the user has basically the permissions to edit a chat.
+    /// The function does not check if the chat type allows editing of concrete elements.
+    pub(crate) async fn can_edit(&self, context: &Context) -> Result<bool> {
```

Comment:
> The name of this function confuses me because can_edit suggests that it's about the ability to change the name and avatar of the chat.

but this is what the function is actually used for :)

it checks the _general_ possibility if a chat can be edited in some way - name, avatar, adding contacts etc. so, the _grant_ in sql-world, if you will.

can_send() then checks if can_edit() is true - plus some additional cornercases.

Reply:
> `can_send()` then checks if `can_edit()` is true - plus some additional cornercases.

It really took me lots of thought to verify this is mostly true in DC, more than everyone who reads this should have to put in I think.
- It's actually rather by chance that this is true for most chats in DC. Other messengers have groups with admins, where you can send messages but not edit the group.
- It's not even true for all chats: You can send to the self-chat (Saved Messages) but you can't edit it.

Reply:
By the way, `can_edit()` probably also returns true for the device chat currently, even though you can't edit it.

What speaks against `is_self_in_chat()`? I think that this would lead to a lot less confusion.

Reply:
k, this is not the hill i want to die on :) i can live with `is_self_in_chat()` as well and, yes, maybe it describes things even in better way. 

i did a commit to change the name.